aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -541,6 +544,110 @@ impl AptosVM {
         RespawnedSession::spawn(self, session_id, resolver, change_set, storage_refund)
     }
 
+    fn execute_signature_verified_transaction(
+        &self,
+        txn: &Transaction,
```

Comment:
The loss of type safety in the interfaces (everywhere) is kinda annoying. Can we do better? Maybe introduce something like this? 
```
struct TxnWithValidSig<'a>(&Transaction)
```

```
impl SignatureCheckedTransaction {
    fn expect_valid(&self) -> TxnWithValidSig {
         ...
    }
}
```

Reply:
Too fancy for me.. I moved this function back to the original call to address your paranoia.. PTAL.